modules/base/src/main/java/org/sandboxpowered/api/entity/Entity.java

```diff
@@ -28,29 +34,51 @@
 
     boolean isSneaking();
 
+    boolean isSubmergedInWater();
```

Comment:
is this specific to water or is it actually submerged in any fluid? if the latter it should be named `isSubmerged`
```suggestion
    boolean isSubmerged();
```

Reply:
I think the vanilla one just checks water, maybe we should have a `isSubmerged(Fluid)` method to be more generic?

Reply:
Could do